lib/message.rb

```diff
@@ -28,7 +28,7 @@ def inflate(json)
       @cc ||= []
       @bcc ||= []
       @labels ||= []
-      @folder ||= nil
+      self.folder ||= nil
```

Comment:
curious, why the `self` here?

Reply:
Writing directly to instance variables in Ruby means there's no seam to inject behavior, so I tend to prefer using attribute writers and readers. This way if we need to do something like say, rename the instance variable, we can leave the old attr reader as an alias and seamlessly upgrade.

Basically it makes it slightly easier to make changes without making changes across the file.

The reason for `self` here is because Ruby thinks that if you do `folder =` you are assigning a local variable, as opposed to using the setter method. Hence `self.folder =`. Python gets around this with an explicit `self` which  (IMO) despite being slightly more verbose is also more user-friendly.